.github/workflows/ci.yml

```diff
@@ -14,14 +14,18 @@ permissions:
 
 jobs:
   test:
-    runs-on: self-hosted
+    runs-on: macos-14
     steps:
       - uses: actions/checkout@v2
+      - name: Setup Xcode Version
+        shell: bash
+        run: |
+          sudo xcode-select --switch /Applications/Xcode_15.4.app
       - name: Run tests on iOS Simulator
         shell: bash
         run: |
-          xcodebuild -scheme BuiltinRegistryGenerator -destination "platform=macOS"
-          xcodebuild test -scheme LiveViewNative-Package -destination "platform=iOS Simulator,name=iPhone 15 Pro" -enableCodeCoverage YES -resultBundlePath TestResults -skipMacroValidation -skipPackagePluginValidation
+          xcodebuild -scheme BuiltinRegistryGenerator -destination "platform=macOS" -derivedDataPath /private/var/tmp/derived_data/
```

Comment:
I'm not sure what is "idiomatic" but in core, [`xcode-build` and then use a github cache to speed it up](https://github.com/liveview-native/liveview-native-core/blob/cb1cf4e591b79e363973b5cb93a4d92adf8dc06c/.github/workflows/ci.yml#L209-L215). I have the different apple targets broken into a matrix so I have like 5 caches on the xcode build.

Reply:
I switched it to match how you're doing it in core. Lmk if it looks correct.

Reply:
LGTM